text_extensions_for_pandas/array/span.py

```diff
@@ -729,16 +836,64 @@ def make_array(cls, o) -> "SpanArray":
             return o
         elif isinstance(o, pd.Series):
             return cls.make_array(o.values)
-        elif isinstance(o, Iterable):
+        elif isinstance(o, Sequence):
             return cls._from_sequence(o)
+        elif isinstance(o, Iterable):
+            return cls._from_sequence([e for e in o])
 
-    @property
-    def target_text(self) -> str:
+    @memoized_property
+    def target_text(self) -> np.ndarray:
         """
-        Returns the common "document" text that the spans in this array
-        reference.
+        :return: "document" texts that the spans in this array reference, as opposed to
+         the regions of these documents that the spans cover.
         """
-        return self._text
+        return self._string_table.ids_to_things(self._text_ids)
+
+    @memoized_property
+    def document_text(self) -> Union[str, None]:
+        """
+        :return: if all spans in this array cover the same document, text of that
+         document. Returns None if the array is emptoy or if teh Spans in this
+          array cover more than one document.
+        """
+        if len(self._text_ids) == 0:
+            return None
+        if not self.is_single_document:
+            return None
```

Comment:
could you return a list of documents for this case? or always return a list, even for one document?

Reply:
I guess that would be the same as `target_text`, but just seems a little strange to return `None` for multiple documents. would it make sense to have
```
@memoized_property
def num_documents()

def document_text(i)
```
to return the `i`th document?

Reply:
Yes, I suppose returning `None` for the multi-document case could lead to trouble. I'll change this property so that it raises an exception if there isn't exactly one document. I'll also change `TokenSpanArray.document_tokens` to match. The purpose of this property is to provide a shorthand for the common pattern, "check that every span in this array is over the same document, then grab that document's text".

You can get a list of all the distinct document texts with:
``` python
(s.document_text for s in my_array.split_by_document())
```
though it might be a good idea to add a dedicated "enumerate just the distinct document texts" method at some point.